Bug 1647447 - Refactor the push information UI

SuyashSalampuria opened the pull request:
![image](https://user-images.githubusercontent.com/44131389/87944325-e00f8f80-cabc-11ea-8fed-1c13b6233d53.png)

I could not find where to find the bug number to be written in the heading if it is not present already 
Also, I have not removed the links to the parent push although they were removed in the recent figma UI
Also, that button showing 6 items at the end of parent revision has not been removed since there were tests written for them already and could not removed without ammending the tests

SuyashSalampuria commented:
@camd please give reviews on this one

camd commented on ui/push-health/CommitHistory.jsx:
```diff
@@ -55,93 +61,121 @@ class CommitHistory extends React.PureComponent {
-                visible={clipboardVisible}
-              />
+          <div className="text-secondary">
+            Push{' '}
```
Instead of having a space with `{' '}`, try adding the class of `ml-1` to the `<span...` on line 84.

`ml-1` is Bootstrap utilities for "margin-left of 1"  I think it might be `1rem`.

camd commented on ui/push-health/CommitHistory.jsx:
```diff
@@ -55,93 +61,121 @@ class CommitHistory extends React.PureComponent {
+              title={expandTitle}
+              aria-label={expandTitle}
+              alt=""
+            />{' '}
```
replace the `{' '}` with `mr-1` on the icon.

Also, let's make the click target be the whole text, not just the icon.  It should also have a `cursor: pointer` to make it more obvious it is doing something.  In this case, you might change the `<span` on line 100 to be a `reactstrap` `<Button>`.  If you add the `outline` param to the `Button` as well as the class of `border-0` then it will work nicely and have the pointer, but won't have the button background.  There are a few examples of me doing this in other areas.

camd commented:
Also, now is a good time to write some unit tests for this and the `<StatusProgress>` Components.  We want the test coverage analysis to be at least at near zero.  But if it could go up, that would be great, too.  :)

camd commented:
@sarah-clements : Would you want to take a look at this PR, too?  The mockup doesn't show any of the commits for the push, and @SuyashSalampuria has followed the mockup very well.  But I seem to recall we agreed to have at least the first two commit messages listed, and the button to "Show all commits" if there were more than two.  Am I remembering that right?

Also, what's your opinion on having the Push summary badge for the Parent Push?  We could leave it out to match the mockup, but it seems like it might be useful.

camd commented:
@SuyashSalampuria I merged your other PR.  Would you rebase this on top of the latest to remove the conflicts?  Thanks

SuyashSalampuria commented:
I did not notice the merge conflicts 
Rebasing it now

sarah-clements commented:
> @sarah-clements : Would you want to take a look at this PR, too? The mockup doesn't show any of the commits for the push, and @SuyashSalampuria has followed the mockup very well. But I seem to recall we agreed to have at least the first two commit messages listed, and the button to "Show all commits" if there were more than two. Am I remembering that right?
> 
Yes, you are correct. There are comments in the figma mockup stating that we should show two commits (comment 3). @SuyashSalampuria you'll need to toggle the comment icon to see comments in the Figma mockups . 

> Also, what's your opinion on having the Push summary badge for the Parent Push? We could leave it out to match the mockup, but it seems like it might be useful.

We could leave it in for now since we're planning to get feedback from users before the soft landing and can get clarity on that feature. However, do we need the clipboard icon to the left on hover? It makes for a strange looking space between "base commit" and the commit number. Maybe it could at least be moved to the end of the commit.

sarah-clements commented:
This looks great overall, just a few comments and questions:

![Screen Shot 2020-07-21 at 12 27 51 PM](https://user-images.githubusercontent.com/19615783/88098179-337dee00-cb4e-11ea-8db2-a5c4bbdb9c29.png)

1. I see two different colors in the pie chart - one for the pending (dark gray) and running (info color). We didn't differentiate between these in the mockup and I'm wondering if we need to. Could we just lump those two into "in progress" and make it a lighter gray? What do you think @camd .

2. When you click on "show all commits" you are repeating the first commit. I think this should be changed to only show commits starting from the third commit. And perhaps changing the text to "show more commits".

3. Per https://github.com/mozilla/treeherder/pull/6644#issuecomment-662063181, the spacing on the base/parent commit looks strange and should be fixed. So maybe we get rid of the copy/paste icon on hover or move it to the end (right side). Also, please change the base commit link to the dark grey/secondary color.

camd commented:
> This looks great overall, just a few comments and questions:
> 
> ![Screen Shot 2020-07-21 at 12 27 51 PM](https://user-images.githubusercontent.com/19615783/88098179-337dee00-cb4e-11ea-8db2-a5c4bbdb9c29.png)
> 
>     1. I see two different colors in the pie chart - one for the pending (dark gray) and running (info color). We didn't differentiate between these in the mockup and I'm wondering if we need to. Could we just lump those two into "in progress" and make it a lighter gray? What do you think @camd .

I agree.  Let's make all in-progress just be a grey color.  Maybe move the copy icon to the end of the line.

SuyashSalampuria commented:
![Screenshot from 2020-07-22 02-55-21](https://user-images.githubusercontent.com/44131389/88108665-d4a88b00-cbc6-11ea-8794-70fa734e82d9.png)
I am not at all sure if it looks the best it can but still I am flagging it for a review @sarah-clements :shipit: 
The clipboard icon is the one which is destroying our UI

sarah-clements commented:
I came up with an expanded version in the Figma mockup that might look a bit better, with the "show/hide" button moved to the bottom. I think making the additional commits more uniform(same font weight and size as the second commit) would look better. Thoughts @camd?

Version 1 - all blue commits, which seems a bit "busy".
![Screen Shot 2020-07-22 at 10 51 38 AM](https://user-images.githubusercontent.com/19615783/88211011-e4958e80-cc09-11ea-8645-96a60dce02a6.png)

Version 2
![Screen Shot 2020-07-22 at 10 55 08 AM](https://user-images.githubusercontent.com/19615783/88211083-01ca5d00-cc0a-11ea-9c87-b881a5541690.png)

SuyashSalampuria commented:
![image](https://user-images.githubusercontent.com/44131389/88287605-26106180-cd10-11ea-9386-acda429c1f47.png)

@sarah-clements I vote for the version 1 just becuase we are using the component Revision.jsx which is used in another component Push.jsx too.
Although the version 2 looks better. But we will need to rewrite the same code just for removing the blue color .

camd commented:
> ![image](https://user-images.githubusercontent.com/44131389/88287605-26106180-cd10-11ea-9386-acda429c1f47.png)
> 
> @sarah-clements I vote for the version 1 just becuase we are using the component Revision.jsx which is used in another component Push.jsx too.
> Although the version 2 looks better. But we will need to rewrite the same code just for removing the blue color .

We can modify `RevisionList.jsx` or `Revision.jsx` to take a param of `commitShaClass` and then apply that class in the Component.  The `defaultProps` can have the blue, but use a class name that is passed in.

camd commented on ui/shared/Revision.jsx:
```diff
@@ -88,7 +89,7 @@ export class Revision extends React.PureComponent {
             title={`Open revision ${commitRevision} on ${repo.url}`}
             href={repo.getRevisionHref(commitRevision)}
-            className="text-monospace commit-sha"
+            className={`text-monospace commit-sha ${commitShaClass}`}
```
Nice!  Yeah, this is exactly what I was talking about.  :)

camd commented on ui/push-health/CommitHistory.jsx:
```diff
@@ -55,94 +61,126 @@ class CommitHistory extends React.PureComponent {
+    const authorEmail = authorMatch ? authorMatch[1] : author;
+    const expandIcon = isExpanded ? faCaretUp : faCaretRight;
+    const expandTitle = isExpanded ? 'Click to collapse' : 'Click to expand';
+    const expandText = isExpanded ? 'Hide all commits' : 'Show more commits';
```
So, this is a bit of a nit-pick, but I wonder if instead of saying "Hide all commits" we used the text: "Hide commits" ?  The reason is that it doesn't actually hide all the commits.  It collapses it down to showing just 1.  :)

sarah-clements commented:
> > ![image](https://user-images.githubusercontent.com/44131389/88287605-26106180-cd10-11ea-9386-acda429c1f47.png)

Looks great :) I have a few nitpicks also:  please change the commits that show when expanded so they look the same as the second commit and the base commit (same font size and weight, as mentioned in my earlier comment). This can be done the same way - with the same custom class or a new one passed in as a prop. And the show/hide commits button text should be a heavier font weight.

SuyashSalampuria commented:
@sarah-clements The base commitshas has a font size of 16px and other revision  sha has a font size of 12 px
Should I change both to 12 px or both to 16 px

sarah-clements commented:
> ![Screenshot from 2020-07-23 23-45-19](https://user-images.githubusercontent.com/44131389/88322624-9f26ad80-cd3e-11ea-872f-e2ad1615eaeb.png)
> @sarah-clements The base commitshas has a font size of 16px and other revision sha has a font size of 12 px
> Should I change both to 12 px or both to 16 px
> Other all changes have been made

I think let's make all the commits 16px. It'll be easier to read.

sarah-clements commented:
> > ![Screenshot from 2020-07-23 23-45-19](https://user-images.githubusercontent.com/44131389/88322624-9f26ad80-cd3e-11ea-872f-e2ad1615eaeb.png)
> > @sarah-clements The base commitshas has a font size of 16px and other revision sha has a font size of 12 px
> > Should I change both to 12 px or both to 16 px
> > Other all changes have been made
> 
> I think let's make all the commits 16px. It'll be easier to read.

Is the revisionSha in the Base commit a different font? It should be the same, so it looks like "Push <sha> in...try".

SuyashSalampuria commented:
> I think let's make all the commits 16px. It'll be easier to read.

I have the comments font to 16 px as well

> Is the revisionSha in the Base commit a different font? It should be the same, so it looks like "Push (revision#) in...try".

In the Revision component, text monospace class has been added to make font type to
```SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace ```
The Base commit has its font type derived from the body as ```-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji ```

sarah-clements commented:
> > I think let's make all the commits 16px. It'll be easier to read.
> 
> I have the comments font to 16 px as well
> 
> > Is the revisionSha in the Base commit a different font? It should be the same, so it looks like "Push (revision#) in...try".
> 
> In the Revision component, text monospace class has been added to make font type to
> `SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace `
> The Base commit has its font type derived from the body as `-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji `

Let's not do monospace, just do the same as what's used for the rest of the text. One font family should be sufficient.

SuyashSalampuria commented:
![image](https://user-images.githubusercontent.com/44131389/88423762-eaf25900-ce09-11ea-900d-5a98a39b3861.png)

@sarah-clements  Seems to be good to go now
I have removed all the redundant classes and remove the class text monospace which was giving the the other font.
However it was needed by one other component, So, added it there.

sarah-clements commented on ui/shared/RevisionList.jsx:
```diff
@@ -16,16 +16,20 @@ export class RevisionList extends React.PureComponent {
 
     return (
-      <Col className={`${widthClass} mb-3`}>
+      <Col className={`${widthClass}`}>
```
This doesn't need to be a template literal anymore.

sarah-clements commented on ui/shared/StatusProgress.jsx:
```diff
@@ -16,11 +16,10 @@ const StatusProgress = (props) => {
           { x: 'failed', y: failed },
         ]}
-        colorScale={['#28a745', '#17a2b8', '#6c757d', '#dc3545']}
+        colorScale={['#28a745', '#6c757d', '#dc3545']}
```
Color `6c757d` should be swapped for the `lightgray` class in the `treeherder-custom-styles` css file.
